stan/math/prim/fun/wiener7_lpdf.hpp

```diff
@@ -0,0 +1,539 @@
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+#ifndef STAN_MATH_PRIM_FUN_WIENER7_LPDF_HPP
+#define STAN_MATH_PRIM_FUN_WIENER7_LPDF_HPP
+
+#include <stan/math/prim/fun/log_sum_exp.hpp>
+#include <stan/math/prim/fun/wiener5_lpdf.hpp>
+
+namespace stan {
+namespace math {
+
+/*
+ * Helper functions for the log of the first passage time density function for a
+ * (Wiener) drift diffusion model with 7 parameters. -boundary separation
+ * (alpha), -drift rate (delta), -relative starting point (beta), -non-decision
+ * time (tau), -inter-trial variability in drift rate (sv). -inter-trial
+ * variability in relative starting point (sw). -inter-trial variability in
+ * non-decition time (st0). For details see wiener_full_lpdf.
+ */
+
+namespace internal {
+
+// tools
+template <typename T_y, typename T_alpha, typename T_delta, typename T_beta,
```

Comment:
One question here:
1) Is there another possibility or common way to combine arguments into a bundle? We implemented the integration-function hcubature in the way that it accepts the function to be integrated and the arguments that are needed for this function in a bundle. Hence, other people can also use hcubature for their application, but we need the arguments as a bunde and cannot hand them over separately. Is maybe Args ...args a better solution? 

hcubature is defined as follows:
`template <typename F, typename T_pars>
double hcubature(const F& integrand, const T_pars& pars, const int& dim,
                 const std::vector<double>& a, const std::vector<double>& b,
                 int& maxEval, const double& reqAbsError,
                 const double& reqRelError) { `

Reply:
Because the `hcubature()` call which requires the struct-packing is only required for the `wiener7`, if you have the standalone `wiener7_lpdf` distribution, then you only need to create this struct of arguments in that distribution function

Reply:
Ok. The structs are now defined in the functions and not as tools outside. Comes in next commit.